compiler/src/dotty/tools/dotc/util/WeakHashSet.scala

```diff
@@ -0,0 +1,410 @@
+/** Taken from the original implementation of WeakHashSet in scala-reflect
```

Comment:
The commit message should include the hash of the version of scalac from which this file was taken.